Doc/howto/functional.rst

```diff
@@ -1,34 +1,24 @@
 ********************************
-  Functional Programming HOWTO
+Functional Programming in Python
 ********************************
 
 :Author: A. M. Kuchling
 :Release: 0.32
 
-In this document, we'll take a tour of Python's features suitable for
-implementing programs in a functional style.  After an introduction to the
-concepts of functional programming, we'll look at language features such as
-:term:`iterator`\s and :term:`generator`\s and relevant library modules such as
-:mod:`itertools` and :mod:`functools`.
```

Comment:
Why remove this paragraph? It seems useful for readers to have a summary at the beginning of what the article's going to cover.

Reply:
A user's eyes will skip over this. They will just bounce off the new reader, for whom they make little connection. They will not add value for the reader who is returning to the page. The links don't take the reader to the topics in the page, but take them away from it, so they don't serve as useful signposts to what's in the page.

If anyone wants to see what's in the page, they can see that immediately from the page's table of contents, where they will find the topics in the context and order of other topics, and can see their place in the hierarchy of concepts.

In general, summaries of what something is going to discuss (and summaries of what has been discussed) add nothing of value to writing. (A summary of conclusions is a quite different matter.)

Reply:
> The links don't take the reader to the topics in the page, but take them away from it, so they don't serve as useful signposts to what's in the page.

I think it's useful to mention `functools` and `itertools` early-on as concrete examples of standard-library modules that allow you to use the functional programming style in Python, especially since the opening section is so abstract. I wouldn't mind it if you changed the links so that they pointed to the passages later on in this document where it explores the `itertools` and `functools` modules.